README.md

```diff
@@ -1511,3 +1516,200 @@ FunctionDeclaration(path) {
 # Best Practices
 
 > I'll be working on this section over the coming weeks.
+
+## Avoid traversing the AST as much as possible
+
+Traversing the AST is expensive, and it's easy to accidentally traverse the AST
+more than necessary. This could be thousands if not tens of thousands of
+extra operations.
+
+Babel optimizes this as much as possible, merging visitors together if it can in
+order to do everything in a single traversal.
+
+### Merge visitors whenever possible
+
+When writing visitors, it may be tempting to call `path.traverse` in multiple
+places where they are logically necessary.
+
+```js
+path.traverse({
+  Identfier(path) {
+    // ...
+  }
+});
+
+path.traverse({
+  BinaryExpression(path) {
+    // ...
+  }
+});
+```
+
+However, it is far better to write these as a single visitor that only gets run
+once. Otherwise you are traversing the same tree multiple times for no reason.
+
+```js
+path.traverse({
+  Identfier(path) {
+    // ...
+  },
+  BinaryExpression(path) {
+    // ...
+  }
+});
+```
+
+### Do not traverse when manual lookup will do
+
+It may also be tempting to call `path.traverse` when looking for a particular
+node type.
+
+```js
+const visitorOne = {
+  Identfier(path) {
+    // ...
+  }
+};
+
+const MyVisitor = {
+  FunctionDeclaration(path) {
+    path.get('params').traverse(visitorOne);
+  }
+};
+```
+
+However, if you are looking for something specific and shallow, there is a good
+chance you can manually lookup the nodes you need without performing a costly
+traversal.
+
+```js
+const MyVisitor = {
+  FunctionDeclaration(path) {
+    path.node.params.forEach(function() {
+      // ...
+    });
+  }
+};
+```
+
+## Optimizing nested visitors
```

Comment:
I feel pedantic even writing this, but maybe `Optimising` instead of `Optimizing` to be consistent with the internal usage in Babel? Ref: https://github.com/babel/babel/pull/2833#discussion-diff-44038468.